### PR TITLE
feat: Add Elasticsearch support

### PR DESCRIPTION
**Pull request by kedod**
### Pull Request Checklist

- [x] New code has 100% test coverage
- [x] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [x] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [x] Pre-Commit Checks were ran and passed
- [x] Tests were ran and passed

### Description

- Add Elasticsearch support

TODO:

- [x] More tests
- [x] Mor accurate tests :)
- [x] Replace tests with async ones

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -4,12 +4,12 @@ Contribution guide
 Setting up the environment
 --------------------------
 
-1. Install `PDM <https://pdm.fming.dev/latest/>`_
-2. Run ``pdm install -G:all`` to create a `virtual environment <https://docs.python.org/3/tutorial/venv.html>`_ and install
-   the dependencies
-3. If you're working on the documentation and need to build it locally, install the extra dependencies with ``pdm install -G:docs``
-4. Install `pre-commit <https://pre-commit.com/>`_
-5. Run ``pre-commit install`` to install pre-commit hooks
+1. Run ``make install-pipx`` to install ``pipx``
```
Comment (JacobCoffee) [inline CONTRIBUTING.rst:7]: 😵

File: src/pytest_databases/docker/elastic_search.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
+# MIT License
```
Comment (JacobCoffee) [inline src/pytest_databases/docker/elastic_search.py:1]: why do we do this on each file?

**Review by JacobCoffee — APPROVED** [review]

File: src/pytest_databases/docker/__init__.py [hunk 3]
```diff
@@ -67,9 +67,7 @@ async def wait_until_responsive(
 
 
 SKIP_DOCKER_COMPOSE: bool = bool(os.environ.get("SKIP_DOCKER_COMPOSE", False))
-USE_LEGACY_DOCKER_COMPOSE: bool = bool(
-    os.environ.get("USE_LEGACY_DOCKER_COMPOSE", os.getenv("GITHUB_ACTIONS") != "true")
-)
+USE_LEGACY_DOCKER_COMPOSE: bool = bool(os.environ.get("USE_LEGACY_DOCKER_COMPOSE", False))
```
[the remark this answers is not in the document]
Comment (cofin) [reply src/pytest_databases/docker/__init__.py:70]: Yeah, I also agree.  I thought i'd flipped this, tbh.

[the remark this answers is not in the document]
Comment (cofin) [reply src/pytest_databases/docker/__init__.py:70]: Let's change it in a separate PR?

File: src/pytest_databases/docker/elastic_search.py (hunk 2 above)
Comment (cofin) [reply src/pytest_databases/docker/elastic_search.py:1]: I've enabled the License required check for ruff.  I have this requirement internally, and I find that it's helpful to be consistent. 

If this is not something we want to do to our repos, then I can remove it.

**Pull request merged**